docs/source/development_guide/ci.rst

```diff
@@ -10,28 +10,89 @@ it implies that the changes introduced by a commit does not introduce any errors
 When creating a PR to master branch or when pushing to master branch, around 35 CI
 tests are run from the following workflows.
 
-#. Tests for DeepChem Core - The jobs are defined in the ``.github/workflows/main.yml`` file. The following jobs are performed in this workflow:
-    * Building and installation of DeepChem in latest Ubuntu OS and Python 3.7 and it checks for ``import deepchem`` 
-    * These tests run on Ubuntu latest version using Python 3.7-3.9 and on windows latest version using Python 3.7. The jobs are run for checking coding conventions using yapf, flake8 and mypy. It also includes tests for doctest and code-coverage.
-    * Tests for pypi-build and docker-build are also include but they are mostly skipped.
+Formatting CI
```

Comment:
Can you add more context here? Explain for each CI what the tests are checking for. Right now, the listing just describes mechanically the steps that are performed and not the reasoning behind what we are trying to do